python/tk_desktop/desktop_window.py

```diff
@@ -681,26 +696,24 @@ def __launch_app_proxy_for_project(self, project, pipeline_configuration_id=None
             self.__set_project_just_accessed(project)
             QtGui.QApplication.instance().processEvents()
 
-            if sys.platform == "darwin":
-                path_field = "mac_path"
-            elif sys.platform == "win32":
-                path_field = "windows_path"
-            elif sys.platform.startswith("linux"):
-                path_field = "linux_path"
-            else:
-                raise SystemError("Unsupported platform: %s" % sys.platform)
-
-            filters = [
-                ["project", "is", project],
-            ]
-
-            fields = [path_field, "users", "code"]
+            # Phase 2: Get information about the pipeline configuration.
 
+            # First get the legacy pipeline configurations, i.e. those using the *_path fields.
```

Comment:
I've been updating the code everywhere calling it classic instead of installed because you could have a pc with hardcoded paths and a plugin id.
so classic means w/ path and w/o plugin id.